### DISCUSSION TITLE
Resonance frequency limitation in lava

### DISCUSSION THREAD

**Discussion by zeinebCh** [lava-nc/lava-dl#181] [Q&A] [answered]
Hello everyone :)

I have a real-valued signal coming out of my radar; the output signal should be in the range of 100 kHz to 600 kHz, and this signal is already in digital format and sampled at 2 MHz to produce 64 samples.

Conventionally, we do an FFT on the samples to get the range information. Instead of doing an FFT, I thought of submitting the 64 samples to 64 resonate neurons, each with a different resonant frequency, and the 64 resonate neurons should cover the bandwidth of the signal from 100 kHz to 600 kHz.

And from our previous discussion, if I distributed the 500 kHz BW among the 64 neurons by assigning a min and max period= [1e-5, 6e-5 ] (min_resonant_frequency=100 kHz, max_resonant_frequency=600 kHz), I will always be getting an error as my period will be smaller than 4.

In accordance, I am willing to get to know if:

Is this scenario in any way feasible in the current version of Lava?
If so, how to bypass the error from a period smaller than 4?
If we have been able to bypass the error from the period, does that mean that the RF neuron is no longer functioning correctly and its oscillation is no longer trustworthy?
I would highly appreciate it if you could provide a small snippet of code showing how I should define a single layer of 64 RF neurons covering the 500 kHz bandwidth, overcoming the error from the period.

Thanks, and I look forward to your response.

**Comment by bamsumit** [comment]
If your period is smaller than 4 time-steps, you will get aliasing in the oscillating dynamics. Thus this limit is enforced and is recommended to use period larger than 4 time-steps to get reasonable results.

**Reply by zeinebCh** [reply]
Thank you very much @bamsumit.
Do you have any recommendation for the period array to cover the 500KHz Bandwidth and overcome the error from the period?
I appreciate a lot your guidance.

**Reply by bamsumit** [reply]
As I said, you must maintain $\text{period} > 4 / f_\text{sampling}$. The exact period depends on the signal you expect to process.

**Reply by zeinebCh** [reply]
@bamsumit Thanks a lot for your help. I appreciate it. 
I just want to verify that I understood your answer correctly. In my case f_sampling=2e6 Hz -> p>8e6 ?

**Reply by bamsumit** [reply, marked as answer]
@zeinebCh there was a typo in my earlier comment. It should have been division, not multiplication. You should understand that these models are discrete-time models. If you has $f_\text{sampling} = 2MHz$, then your discrete $\text{time per step} = 0.5\mu s$. So if you set your discrete $\text{period} = 4$, that is equivalent to setting a period of $2\mu s$ in continuous domain and equivalently a resonant frequency of $500 KHz$.

**Reply by zeinebCh** [reply]
That is a great help for me. Thank you very much.
